runtime/onert/core/src/exec/ExecutionObservers.cc

```diff
@@ -146,61 +142,43 @@ TracingObserver::~TracingObserver()
 
 void TracingObserver::handleSubgraphBegin(ir::SubgraphIndex subg_ind)
 {
-  // TODO Write subg_ind into profling result
-  UNUSED_RELEASE(subg_ind);
-  _collector.onEvent(EventCollector::Event{EventCollector::Edge::BEGIN, "runtime", "Graph"});
+  _collector.onEvent(
+    EventCollector::SubgEvent{_tracing_ctx, EventCollector::Edge::BEGIN, subg_ind.value()});
 }
 
 void TracingObserver::handleJobBegin(IExecutor *, ir::SubgraphIndex subg_ind,
                                      const ir::OpSequence *op_seq, const backend::Backend *backend)
 {
-  // TODO Write subg_ind into profling result
-  UNUSED_RELEASE(subg_ind);
+  if (op_seq->size() == 0)
+    throw std::runtime_error{"Empty OpSequence"};
+  const auto &first_op_idx = op_seq->operations().at(0);
 
   std::string backend_id = backend->config()->id();
-
-  auto ev = EventCollector::Event{EventCollector::Edge::BEGIN, backend_id,
-                                  opSequenceTag(op_seq, _graph.operations())};
+  auto ev = EventCollector::OpEvent{
+    _tracing_ctx, EventCollector::Edge::BEGIN, subg_ind.value(), backend_id, first_op_idx.value(),
+    _graph.operations().at(first_op_idx).name(), op_seq->size()};
   // add shape of inputs
   setUserData(_graph, op_seq, ev.userData);
-
   _collector.onEvent(ev);
 }
 
 void TracingObserver::handleJobEnd(IExecutor *, ir::SubgraphIndex subg_ind,
                                    const ir::OpSequence *op_seq, const backend::Backend *backend)
 {
-  // TODO Write subg_ind into profling result
-  UNUSED_RELEASE(subg_ind);
+  if (op_seq->size() == 0)
+    throw std::runtime_error{"Empty OpSequence"};
+  const auto &first_op_idx = op_seq->operations().at(0);
 
   std::string backend_id = backend->config()->id();
-  _collector.onEvent(EventCollector::Event{EventCollector::Edge::END, backend_id,
-                                           opSequenceTag(op_seq, _graph.operations())});
+  _collector.onEvent(EventCollector::OpEvent{
+    _tracing_ctx, EventCollector::Edge::END, subg_ind.value(), backend_id, first_op_idx.value(),
+    _graph.operations().at(first_op_idx).name(), op_seq->size()});
 }
 
 void TracingObserver::handleSubgraphEnd(ir::SubgraphIndex subg_ind)
 {
-  // TODO Write subg_ind into profling result
-  UNUSED_RELEASE(subg_ind);
-
-  _collector.onEvent(EventCollector::Event{EventCollector::Edge::END, "runtime", "Graph"});
-}
-
-std::string TracingObserver::opSequenceTag(const ir::OpSequence *op_seq,
```

Comment:
Removed since this is meaningful when making the name of op for chrome tracing. Same feature was added into ChromeTracingEventWriter.